universal-hash/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+//! Traits for [Universal Hash Functions].
+//!
+//! Universal hash functions select from a "universal family" of possible
+//! hash functions selected by a key. They are well suited to the purpose
+//! of "one time authenticators" for a sequence of bytestring inputs,
+//! as their construction has a number of desirable properties such as
+//! pairwise independence as well as amenability to efficient implementations,
+//! particularly when implemented using SIMD instructions.
+//!
+//! When combined with a cipher, such as in Galois/Counter Mode or the
+//! Salsa20 family AEAD constructions, they can provide the core functionality
+//! for a Message Authentication Code (MAC).
+//!
+//! [Universal Hash Functions]: https://en.wikipedia.org/wiki/Universal_hashing
+
+#![no_std]
+#![doc(html_logo_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo_small.png")]
+#![warn(missing_docs, rust_2018_idioms)]
+
+#[cfg(feature = "std")]
+extern crate std;
```

Comment:
BTW maybe add `pub extern crate generic_array;` like we do in other trait crates?

Reply:
Oops, I had that but it got lost in the 2018 edition upgrade. Will do a `pub use generic_array`

Reply:
IIRC `pub extern` works on 2018 edition. I think using `pub use` will not be detected by rustdoc as a [re-export](https://docs.rs/crypto-mac/0.7.0/crypto_mac/#reexports).

Reply:
`pub extern` is disallowed by the `rust_2018_idioms` lint:

<img width="625" alt="Screen Shot 2019-08-30 at 11 49 44 AM" src="https://user-images.githubusercontent.com/797/64044736-6a08df80-cb1c-11e9-8f67-5e2f96937cbf.png">

`pub use` is detected as a re-export by rustdoc:

<img width="240" alt="Screen Shot 2019-08-30 at 11 49 19 AM" src="https://user-images.githubusercontent.com/797/64044745-73924780-cb1c-11e9-8995-f5086d2e7176.png">

Reply:
Ah, good to know.